### PR TITLE
fix: can settings username and email for checkout

### PR DESCRIPTION
**Pull request by wahapo**
Add environment variables `SCM_USERNAME` and `SCM_EMAIL` with use custom user for checkout.
https://github.com/screwdriver-cd/scm-github/pull/52
https://github.com/screwdriver-cd/scm-bitbucket/pull/34

As a further modification, if a user does not have a valid SCM token registered with Screwdriver, `SCM_USERNAME` will use this user's token instead.

### CODE REVIEW

File: config/default.yaml [hunk 1]
```diff
@@ -116,6 +116,9 @@ scm:
         oauthClientSecret: AGAIN-SOMETHING-HERE-IS-USEFUL
         # You can also configure for use with GitHub enterprise
         #gheHost: github.screwdriver.cd
+        # The username and email used for checkout with github
+        #username: sd-buildbot
```
Comment (stjohnjohnson) [inline config/default.yaml]: I wouldn't comment these out, let them be there

File: plugins/webhooks/index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 const boom = require('boom');
+const joi = require('joi');
 const GENERIC_SCM_USER = 'sd-buildbot';
```
Comment (stjohnjohnson) [inline plugins/webhooks/index.js]: I would remove this, as you can default it in the default.yaml

File: config/default.yaml [hunk 3]
```diff
@@ -125,6 +128,13 @@ scm:
     bitbucket:
         oauthClientId: YOUR-BITBUCKET-OAUTH-CLIENT-ID
         oauthClientSecret: YOUR-BITBUCKET-OAUTH-CLIENT-SECRET
+        # The username and email used for checkout with bitbucket
+        #username: sd-buildbot
+        #email: dev-null@screwdriver.cd
+
+webhooks:
```
Comment (stjohnjohnson) [inline config/default.yaml]: I think you need to load this in the `/bin/server.js`

File: config/default.yaml [hunk 4]
```diff
@@ -125,6 +128,13 @@ scm:
...
+    # Obtains the SCM token for a given user. If a user does not have a valid SCM token registered with Screwdriver, it will use this user's token instead.
+    #username: sd-buildbot
```
Comment (stjohnjohnson) [inline config/default.yaml]: I wouldn't leave this commented out

File: config/default.yaml (hunk 1 above)
Comment (d2lam) [inline config/default.yaml]: Shouldn't matter though...

File: config/default.yaml (hunk 3 above)
Comment (wahapo) [inline config/default.yaml]: Thank you for pointing out. Added.

File: config/default.yaml (hunk 4 above)
Comment (wahapo) [inline config/default.yaml]: modified

File: plugins/webhooks/index.js (hunk 2 above)
Comment (wahapo) [inline plugins/webhooks/index.js]: removed.

File: config/default.yaml (hunk 1 above)
Comment (wahapo) [inline config/default.yaml]: Modified.

**Pull request merged**